app/services/support_interface/expanded_qualifications_export.rb

```diff
@@ -0,0 +1,48 @@
+module SupportInterface
+  class ExpandedQualificationsExport
+    def data_for_export
+      application_choices = ApplicationChoice
+        .select(:id, :application_form_id, :status, :course_option_id, :rejection_reason, :structured_rejection_reasons)
+        .includes(:course_option, :course, :provider, application_form: [:application_qualifications])
+        .order(:application_form_id)
+
+      application_choices.find_each(batch_size: 100).lazy.flat_map do |application_choice|
+        application_form = application_choice.application_form
+        course = application_choice.course
+        qualifications = application_form.application_qualifications
+
+        qualifications.map do |qualification|
+          {
+            application_form_id: application_form.id,
+            choice_id: application_choice.id,
+            qualification_id: qualification.id,
+            candidate_id: application_form.candidate_id,
+            support_reference: application_form.support_reference,
+            phase: application_form.phase,
+            recruitment_cycle_year: application_form.recruitment_cycle_year,
+
+            choice_status: application_choice.status,
+            rejection_reason: application_choice.structured_rejection_reasons || application_choice.rejection_reason,
+            course_code: course.code,
+            provider_code: course.provider.code,
+
+            level: qualification.level,
+            qualification_type: qualification.qualification_type,
+            other_uk_qualification_type: qualification.other_uk_qualification_type,
+            award_year: qualification.award_year,
+            subject: qualification.subject,
+            predicted_grade: qualification.predicted_grade,
+            grade: qualification.grade,
+            institution_name: qualification.institution_name,
+            institution_country: qualification.institution_country,
+            international: qualification.international,
+            comparable_uk_degree: qualification.comparable_uk_degree,
+            non_uk_qualification_type: qualification.non_uk_qualification_type,
+            comparable_uk_qualification: qualification.comparable_uk_qualification,
+            constituent_grades: qualification.constituent_grades,
```

Comment:
wondering if these could be grouped slightly more intuitively. Maybe `constituent_grades` should be on line 36? and we could have `non_uk_qualification_type ` on line 37 followed by all the related attrs like `institution_country ` and `comparable_uk_degree` etc.

Reply:
yep, definitely - have pushed a change

Reply:
Great. Looks to me 👍